src/txn/txn_timestamp.c

```diff
@@ -946,6 +946,31 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     return (0);
 }
 
+/*
+ * __wt_txn_set_timestamp_commit --
+ *     Directly set the commit timestamp in a transaction, bypassing parsing logic. Prefer this to
+ *     @ref __wt_txn_set_timestamp when string parsing is a performance bottleneck.
+ */
+int
+__wt_txn_set_timestamp_commit(WT_SESSION_IMPL *session, wt_timestamp_t ts)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+
+    WT_TRET(__wt_txn_context_check(session, true));
+
+    WT_RET(__wt_txn_set_commit_timestamp(session, ts));
```

Comment:
Can you check against WT_TS_NONE (=0), if that value is passed as an argument, we want to not set it. It will be useful to add a python test around it too.

Reply:
I've added a check in 266fdf1, but I'm having a little trouble adding a test for it. The function now (correctly) ignores a commit timestamp of zero, but calling `transaction_commit()` seems to use some reasonable default when a commit timestamp isn't supplied, so the test value will get inserted. (I can't supply `transaction_commit()` a timestamp of zero since that raises a different error.) Do you have any suggestions around this?

Reply:
The zero check is now done in a shared path, but I can't see any of the existing tests checking what happens if zero is passed in. Would you say writing these in addition to the existing tests in this PR is in scope?

Reply:
Added some zero tests in e8e7c4d